File: binarySearch.cpp
int search(vector<int>& nums, int target) {

    int mid = nums.size()/2; // mid=3
    int L = 0;
    int R = nums.size()-1; // R=5

    while(L <= R) { // 0 < 5; 3 < 5

        // nums[3] = 5; target = 9; nums[4]=9; target == 9
        if(nums[mid] == target) {
            return mid;
        }
        else if(nums[mid] < target) {
            L = mid+1; // L=4
        }
        else {
            R = mid-1;
        }
        mid=(L+R)/2;
    }
    return -1;


}

File: python/binarySearch.py
class Solution:
    def search(self, nums: List[int], target: int) -> int:
        l = 0
        r = len(nums)-1
        while l <= r:
            mid = int((l+r)/2)
            print(mid)
            if(nums[mid] == target):
                return mid
            elif nums[mid] > target:
                # Go left
                r = mid-1
            else:
                l = mid+1

        return -1
